src/main/java/com/solace/samples/BasicRequestor.java

```diff
@@ -34,98 +38,115 @@
 
-        // Send the request then wait for a reply
-        producer.send(requestDestination, request, DeliveryMode.NON_PERSISTENT, Message.DEFAULT_PRIORITY,
+        // Send the request
+        requestProducer.send(requestTopic, request, DeliveryMode.NON_PERSISTENT,
+                Message.DEFAULT_PRIORITY,
                 Message.DEFAULT_TIME_TO_LIVE);
-        Message reply = consumer.receive(timeoutMs);
-
-        if (reply == null) {
-            System.out.println("Failed to receive a reply in " + timeoutMs + " msecs");
-            return;
-        }
 
-        if (reply.getJMSCorrelationID() == null) {
-            throw new Exception(
-                    "Received a reply message with no correlationID.  This field is needed for a direct request.");
-        }
+        System.out.println("Sent successfully. Waiting for reply...");
 
-        // The reply's correlationID should match the request's correlationID
-        if (!reply.getJMSCorrelationID().equals(correlationId)) {
-            throw new Exception("Received invalid correlationID in reply message.");
-        }
+        // the main thread blocks at the next statement until a message received or the timeout occurs
+        Message reply = replyConsumer.receive(REPLY_TIMEOUT_MS);
 
-        // Process the reply
-        if (reply instanceof TextMessage) {
-            System.out.printf("TextMessage response received: '%s'%n", ((TextMessage) reply).getText());
-            if (!reply.getBooleanProperty(SupportedProperty.SOLACE_JMS_PROP_IS_REPLY_MESSAGE)) {
-                System.out.println("Warning: Received a reply message without the isReplyMsg flag set.");
+        if (reply == null) {
+            System.out.println("Failed to receive a reply in " + REPLY_TIMEOUT_MS + " msecs");
+        } else {
+            if (reply.getJMSCorrelationID() == null) {
+                System.out.println(
+                        "Received a reply message with no correlationID. This field is needed for a direct request.");
+            } else {
+                if (!reply.getJMSCorrelationID().replaceAll("ID:", "").equals(correlationId)) {
+                    System.out.println("Received invalid correlationID in reply message.");
+                } else {
+                    if (reply instanceof TextMessage) {
+                        System.out.printf("TextMessage response received: '%s'%n", ((TextMessage) reply).getText());
+                    } else {
+                        System.out.println("Message response received.");
+                    }
+                    if (!reply.getBooleanProperty(SupportedProperty.SOLACE_JMS_PROP_IS_REPLY_MESSAGE)) {
+                        System.out.println("Warning: Received a reply message without the isReplyMsg flag set.");
+                    }
+                    System.out.printf("Message Content:%n%s%n", SolJmsUtility.dumpMessage(reply));
```

Comment:
This printout has been made redundant by the above code. Suggest removing it or reverting the code.